airflow/contrib/hooks/azure_hdinsight_hook.py

```diff
@@ -0,0 +1,252 @@
+            response = requests.get(
+                url,
+                params=self.params,
+                headers=self.headers,
+                auth=(self.username, self.password))
+        else:
+            response = requests.post(
+                url,
+                data=data,
+                auth=(self.username, self.password),
+                headers=self.headers
+            )
+        # logging.debug("Sending to %s  %s", response.request)
+        if response.status_code in acceptable_response_codes:
+            return response.json()
+        else:
+            result = "Statement  Returned \'{0}\'." \
+                .format(str(response.text))
+            logging.error(result)
+            raise AirflowException(result)
+
+    def _get_job_status(self, endpoint):
+        return self._http_rest_call(method="GET", endpoint=endpoint)
+
+    def create_cluster(self,
+                       template_json,
+                       parameters_json,
+                       deploy_name
+                       ):
+        """
+        References
+        https://docs.microsoft.com/en-us/python/api/overview/azure/hdinsight?view=azure-python
+
+        :return:
+        """
+        client = ResourceManagementClient(self._get_credential(), self.subscription_id)
+        """Deploy the template to a resource group."""
+        client.resource_groups.get(
+            self.resource_group_name,
+            self.resource_group_location
+        )
+
+        deployment_properties = DeploymentProperties(
+            mode=DeploymentMode.incremental,
+            template=template_json,
+            parameters=parameters_json)
+
+        deployment_async_operation = client.deployments.create_or_update(
+            self.resource_group_name,
+            deploy_name,
+            deployment_properties
+        )
+        deployment_async_operation.wait()
+        return deployment_async_operation.result()
+
+    def delete_cluster(self):
+        """
+        References
+        https://docs.microsoft.com/en-us/python/api/overview/azure/hdinsight?view=azure-python
+
+        :return:
+        """
+        client = HDInsightManagementClient(self._get_credential(), self.subscription_id)
+        client.clusters.create()
+        delete_poller = client.clusters.delete(self.resource_group_name,
+                                               cluster_name=self.cluster_name)
+        delete_poller.wait()
+        return delete_poller.result()
+
+    def _get_credential(self) -> ServicePrincipalCredentials:
+        """
+        Credential of Azure App Registration
+
+        References
+        https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal
+        https://docs.microsoft.com/en-us/python/api/overview/azure/key-vault?toc=%2Fpython%2Fazure%2FTOC.json&view=azure-python
+
+        :rtype: ServicePrincipalCredentials
+        """
+        return ServicePrincipalCredentials(
```

Comment:
Other hooks for Azure accepts a credentials via `AZURE_AUTH_LOCATION`.
https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/azure_container_instance_hook.py#L63
I do not know if it is necessary to add support here or delete support in another place. I believe that support for Azure should be uniform.